frontend/app/src/components/history/events/forms/EthDepositEventForm.vue

```diff
@@ -0,0 +1,361 @@
+      requiredIf(() => !!get(editableItem))
+    )
+  },
+  amount: {
+    required: helpers.withMessage(
+      t('transactions.events.form.amount.validation.non_empty').toString(),
+      required
+    )
+  },
+  usdValue: {
+    required: helpers.withMessage(
+      t('transactions.events.form.fiat_value.validation.non_empty', {
+        currency: get(currencySymbol)
+      }).toString(),
+      required
+    )
+  },
+  sequenceIndex: {
+    required: helpers.withMessage(
+      t(
+        'transactions.events.form.sequence_index.validation.non_empty'
+      ).toString(),
+      required
+    )
+  },
+
+  validatorIndex: {
+    required: helpers.withMessage(
+      t(
+        'transactions.events.form.validator_index.validation.non_empty'
+      ).toString(),
+      required
+    )
+  },
+  depositor: {
+    required: helpers.withMessage(
+      t('transactions.events.form.depositor.validation.non_empty').toString(),
+      required
+    ),
+    isValid: helpers.withMessage(
+      t('transactions.events.form.depositor.validation.valid').toString(),
+      (value: string) => isValidEthAddress(value)
+    )
+  }
+};
+
+const { setValidation, setSubmitFunc } = useHistoryEventsForm();
+
+const v$ = setValidation(
+  rules,
+  {
+    timestamp: datetime,
+    eventIdentifier,
+    txHash,
+    amount,
+    usdValue,
+    sequenceIndex,
+    validatorIndex,
+    depositor
+  },
+  {
+    $autoDirty: true,
+    $externalResults: errorMessages
+  }
+);
+
+const reset = () => {
+  set(sequenceIndex, get(nextSequence) || '0');
+  set(txHash, '');
+  set(eventIdentifier, null);
+  set(datetime, convertFromTimestamp(dayjs().unix()));
+  set(amount, '0');
+  set(usdValue, '0');
+  set(validatorIndex, '');
+  set(depositor, '');
+  set(extraData, {});
+  set(errorMessages, {});
+
+  get(assetPriceForm)?.reset();
+};
+
+const applyEditableData = async (entry: EthDepositEvent) => {
+  set(sequenceIndex, entry.sequenceIndex?.toString() ?? '');
+  set(txHash, entry.txHash);
+  set(eventIdentifier, entry.eventIdentifier);
+  set(datetime, convertFromTimestamp(entry.timestamp));
+  set(amount, entry.balance.amount.toFixed());
+  set(usdValue, entry.balance.usdValue.toFixed());
+  set(validatorIndex, entry.validatorIndex.toString());
+  set(depositor, entry.locationLabel);
+  set(extraData, entry.extraData || {});
+};
+
+const applyGroupHeaderData = async (entry: EthDepositEvent) => {
+  set(sequenceIndex, get(nextSequence) || '0');
+  set(eventIdentifier, entry.eventIdentifier);
+  set(txHash, entry.txHash);
+  set(validatorIndex, entry.validatorIndex.toString());
+  set(depositor, entry.locationLabel ?? '');
+  set(datetime, convertFromTimestamp(entry.timestamp));
+};
+
+const { setMessage } = useMessageStore();
+
+const { editHistoryEvent, addHistoryEvent } = useHistoryTransactions();
+
+const save = async (): Promise<boolean> => {
+  const timestamp = convertToTimestamp(get(datetime));
+
+  const payload: Writeable<NewEthDepositEventPayload> = {
```

Comment:
same here if we don't modify the payload after the assignment no need for it to be writable afterwards